spec/immutable-data-structures.html

```diff
@@ -6,27 +6,590 @@ <h1>Immutable Data Structures</h1>
+      </ul>
       <emu-clause id="sec-record-constructor-record-value">
-        <h1>Record ( _value_ )</h1>
-        <p>TBD</p>
+        <h1>Record ( _arg_ )</h1>
+        <p>When the `Record` function is called, the following steps are taken:</p>
+        <emu-alg>
+          1. If NewTarget is not *undefined*, throw a *TypeError* exception.
+          1. Let _obj_ be ? ToObject(_arg_).
+          1. Let _props_ be ? EnumerableOwnPropertyNames(_obj_, ~key+value~).
+          1. Let _fields_ be a new empty List.
+          1. For each element _prop_ of _props_,
+            1. Let _name_ be ! GetV(_prop_, `"0"`).
+            1. Let _value_ be ! GetV(_prop_, `"1"`).
+            1. If Type(_value_) is Object, throw a *TypeError* exception.
+            1. Let _field_ be { [[Key]]: _name_, [[Value]]: _value_ }.
+            1. Append _field_ to the end of list _fields_.
+          1. Return a new Record value whose [[Fields]] is _fields_.
+        </emu-alg>
+      </emu-clause>
+    </emu-clause>
+    <emu-clause id="sec-properties-of-the-record-constructor">
+      <h1>Properties of the Record Constructor</h1>
+      <p>The Record constructor:</p>
+      <ul>
+        <li>has a [[Prototype]] internal slot whose value is %Record.prototype%.</li>
+        <li>has the following properties:</li>
+      </ul>
+      <emu-clause id="sec-record.fromentries">
+        <h1>Record.fromEntries ( _iterable_ )</h1>
+        <p>The *fromEntries* function takes one argument _iterable_, and performs the following steps:</p>
+        <emu-alg>
+          1. Perform ? RequireObjectCoercible(_iterable_).
+          1. Let _fields_ be a new empty List.
+          1. Let _iteratorRecord_ be ? GetIterator(_iterable_).
+          1. Let _stepsDefine_ be the algorithm steps defined in <emu-xref href="#sec-append-record-field-to-list-functions" title></emu-xref>.
+          1. Let _adder_ be ! CreateBuiltinFunction(_stepsDefine_, &laquo; &raquo;).
+          1. Perform ! AddEntriesFromIterable(_fields_, _iterable_, _adder_).
+          1. Return a new Record value whose [[Fields]] is _fields_.
+        </emu-alg>
+        <emu-note>
+          <p>The parameter _iterable_ is expected to be an object that implements an @@iterator method that returns an iterator object that produces a two element array-like object whose first element is a value that will be used as a Map key and whose second element is the value to associate with that key.</p>
+        </emu-note>
+      </emu-clause>
+      <emu-clause id="sec-append-record-field-to-list-functions">
+        <h1>AppendRecordFieldToList Functions</h1>
+        <p>An AppendRecordFieldToList function is an anonymous built-in function. When an AppendRecordFieldtoList function is called with arguments _key_ and _value_, the following steps are taken:</p>
+        <emu-alg>
+            1. Let _list_ be the *this* value.
+            1. Assert: Type(_list_) is List.

```

Comment:
List isn't a JS type, and Type() can never return "List".

I'm not sure what to do here; the receiver of a function (even a spec artifice) has to be a JS value.

Reply:
Agree that the receiver can't be a spec List, but there should be a ton of ways to fix this depending on what you're trying to do. I'm not sure what to do either, though, as I don't see any callers of this function.

Reply:
Oops, I see the cross-reference is by id

Reply:
Rather than threading the fields through the `this` value, I suggest using *undefined* for the `this` value, and use an [abstract closure](https://tc39.es/ecma262/#sec-abstract-closure) to capture the _fields_ list.